#463 Restructuring of documentation TOC

jrudolph opened the pull request:
Refs #436 

The idea is to start restructuring the top-level TOC to give the most important topics the best spots. The "chapter numbers" are mostly a cosmetic feature that feels to give the docs an ordered appearance (and hopefully prevents adding more entries lightly). I also added some other smaller "features".

![screenshot from 2017-12-22 16 07 19](https://user-images.githubusercontent.com/9868/34302670-3737f980-e732-11e7-835b-a123a619173f.png)

2m commented on docs/src/main/paradox/scala/http/06extensions.md:
```diff
@@ -0,0 +1,3 @@
+# 6. Extensions ➕
```
The extensions icon does not render here. Neither Chromium, nor IntelliJ. That is a rare plus you found. :)

Nevertheless, I like the icons. :)

ktoso commented on docs/src/main/paradox/scala/http/06extensions.md:
```diff
@@ -0,0 +1,3 @@
+# 6. Extensions
+
+This page list well-known third-part extensions of Akka HTTP. See the Akka HTTP tag on scaladex for more info. // TODO: add link 
```
Link to https://akka.io/community/  

or https://index.scala-lang.org/search?topics=akka-http

ktoso commented:
Release notes last? Or do something so they don't continue growing into forever in the TOC?

ktoso commented:
Not sure the utf characters make sense though hm, looks a bit silly

jrudolph commented:
> Release notes last? Or do something so they don't continue growing into forever in the TOC?

Yep, I guess at some point we should fold them into subpages like "10.0.x release note" etc.

> Not sure the utf characters make sense though hm, looks a bit silly

I removed them again for now. We can still add them later on if we (unlikely enough) happen to agree on something. ;)

ktoso commented:
> I removed them again for now. We can still add them later on if we (unlikely enough) happen to agree on something. ;) 

ok, sorry ;-)

jonas commented:
I'm not sure about the Dingbats. It's an extremely limited "icon set" and many of the proposed icons give no clue at all, which IMO makes them pointless. For example, security often uses a shield icon, extensions/plugins a jigsaw puzzle piece.

jrudolph commented:
@jonas, I see. Let's not go there for now.

ktoso commented on docs/src/main/paradox/scala/http/routing-dsl/index.md:
```diff
@@ -1,4 +1,4 @@
-# High-level Server-Side API
+# Routing DSL
```
Oh yeah great rename

ktoso commented on docs/src/main/paradox/scala/http/server-side/low-level-api.md:
```diff
@@ -1,6 +1,6 @@
-# Low-Level Server-Side API
+# Core Server API
```
Also good rename, I always thought not many people get the "low / high level" wording we used

ktoso commented on docs/src/main/paradox/scala/http/security.md:
```diff
@@ -1,4 +1,4 @@
-# Security Announcements
+# ! Security Announcements !
```
good ;) I can live with that ;)

jrudolph commented:
A few more changes here:

 * removed the numbering of top-level files
 * copied contributing.md into contributing section, I think it's nice to have that inside the documentation (not sure how to handle that duplication, probably we could link to the documentation from the contributing.md)
 * A bit of page title cleanup (Server-side API => Server API, Low-level Server-Side API => Core Server API, High-level server-side API => Routing DSL)
 * Filled out technologies page
 * Moved some sections under "Tips And Tricks"
 * Reference section that links to reference information at the bottom

jrudolph commented:
Solved conflicts.
